src/macros.rs

```diff
@@ -1,12 +1,26 @@
 //! Macros used on ouch.
 
 /// Macro that prints \[INFO\] messages, wraps [`println`].
+///
+/// Normally, this prints nothing if ACCESSIBLE mode is turned on,
+/// except when called as `info!(a11y_show, "..", ..)`
 #[macro_export]
 macro_rules! info {
-    ($($arg:tt)*) => {
-        $crate::macros::_info_helper();
+    // Show info message even in ACCESSIBLE mode
+    (a11y_show, $($arg:tt)*) => {
```

Comment:
I'm not that happy with this `info!(a11y_show, ...)` and it confused others as well. The idea is to somehow mark some `info!()` calls as relevant for accessibility-mode output whereas all other `info!()` calls without `a11y_show` are suppressed. I'm struggling to come up with a better way to handle this, do you have any ideas what we could do here or whether this is okay as long as it's properly documented?

Reply:
Oops, saw this earlier today, but forgot to comment.

I also think thought at first that the macro is a bit confusing but I didn't knew how to improve it, so I just accepted it.

Now that you asked again, I think I came with an idea.

What about having:

```rust
info!(always, "Message here");
```

And:

```rust
info!(not_tts, "Message here");
```

The `always` one is always displayed, and the `not_tts` is disabled when the user might be using `tts`.

So then the person who is using `info!` will always have to keep in mind "is this message tts-friendly?

The name does not seem right, so here's another idea:

```rust
info!(accessible, "Message Here");
info!(inaccessible, "Message Here");
```

Accessible messages can be displayed every time, inaccessible messages are flood (too much text on screen).

We could have another macro, but, this will certainly lead to people not worrying about accessibility, forgetting to use the correct one.

Additionally, for this to cause less confusion, we'd need explanation in the docs on why some messages are hidden, not everyone knows what TTS tools are, and those people won't figure out that it's because too much text makes it hard to navigate and understand the output (if I'm not mistaken, that's the reason why we are using this macro versions).

Reply:
Neat, I think I prefer the second idea with _accessible_ and _inaccessible_. I think the term TTS may be unfamiliar to many devs..

Reply:
I'm almost done, just one `info!` remains that I'm unsure about:
https://github.com/ouch-org/ouch/blob/cb6d65c79aa917e367c994482c760682e97d8fbb/src/archive/zip.rs#L152
I am not very familiar with ZIP files and neither the function declaration nor the call in L.47 has any documentation/comments. Is this an important message?

Apart from that, I don't understand this even from a seeing person's viewpoint. What does it mean? What is a ZIP archive comment, why is this message printed?

Reply:
Someone can insert a "comment" in the metadata of this zip file, telling where it comes from, or what it is, or a link, a name, anything.

Is one of the features of zip archives, like passwords (that we don't support yet because ... ).

I think we should display it, if a person inserts a book in the comment, our tool will not be accessible anymore, but it can be bad to hide this info.

Googled for "zip comment size limit":

> The .ZIP file format allows for a comment containing up to 65,535 (216−1) bytes of data to occur at the end of the file after the central directory.

It might contain thousands of words, or just one very useful piece of information.

I never see a comment in zip archives, so I don't have a very strong opinion on this, you can decide.

It's a tradeoff: show a possibly useful information or making sure our application is 100% accessible.

Both are reasonable choices.

Reply:
Neat, thanks for looking that up.

I marked it as accessible for now and included a comment stating that if we stumble across very long, unnecessary comments in the future, it might be sensible to ask the user first before printing the whole comment (showing the length of the comment beforehand).

But until then, I think just printing it is good enough